### DISCUSSION TITLE
Configuration example for using the auto-configuration approach

### DISCUSSION THREAD

**Discussion by anadimisra** [flamingock/flamingock-java#641] [General]
While the documentation mentions an autoconfigure approach as well for runner, which is an alternative to building the runner, it's not clear how to activate the autoconfigure for Mongock. I'm passing the following properties, however my app fails to create an application runner and obviously no migrations run.

```properties
        "mongock:runner-type=ApplicationRunner",
        "mongock:driver=MongoReactiveDriver",
        "mongock.migration-scan-package=com.digite.cloud.blox.entity.seed",
        "mongock.transaction-enabled=true",
        "mongock.lock-acquired-for-millis=3000",
        "mongock.lock-quit-trying-after-millis=100",
        "mongock.lock-try-frequency-millis=50",
        "mongock.index-creation=true",
        "mongock.write-concern={w:W1,wTimeoutMs:1000,j:true}",
        "mongock.read-concern=majority",
        "mongock.read-preference=primaryPreferred"
```

dependencies in my maven POM file.

```xml
        <dependency>
            <groupId>io.mongock</groupId>
            <artifactId>mongock-springboot</artifactId>
        </dependency>
        <dependency>
            <groupId>io.mongock</groupId>
            <artifactId>mongodb-reactive-driver</artifactId>
        </dependency>
```
When I add `@EnableMongock` annotation to the main spring boot application class I get the error 

` Parameter 0 of method getBuilder in io.mongock.runner.springboot.config.MongockContext required a bean of type 'io.mongock.driver.api.driver.ConnectionDriver' that could not be found. `

**Comment by osantana85** [comment]
Hello @anadimisra , Mongock currently doesn't support auto-configuration with MongoDB Reactive Driver. You should use the builder approach.

For more information, please take a look to our example: [Mongock with Springboot + MongoDB Reactive Driver](https://github.com/mongock/mongock-examples/tree/master/mongodb/springboot-reactive)

And the [documentation](https://docs.mongock.io/v5/driver/mongodb-reactive/index.html)

Thanks for using Mongock.
